automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -817,9 +834,201 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+    for (c <- complete) {
+      val newInputs = c.arguments("input").filter(m => !m.label.contains("Unit") && !m.text.contains("self") && !m.tags.get.head.contains("VB"))
+      val newOutputs = c.arguments("output").filter(m => !m.label.contains("Unit") && !m.text.contains("self") && !m.tags.get.head.contains("VB"))
+      if (newInputs.nonEmpty && newOutputs.nonEmpty) {
+        val newArgs = Map("input" -> newInputs, "output" -> newOutputs)
+        val newFunctions = copyWithArgs(c, newArgs)
+        toReturn.append(newFunctions)
+      }
+    }
+    for (f <- fragment) {
+      if (f.arguments.contains("input")) {
+        val inputFilter = f.arguments("input").filter(!_.label.contains("Unit") && f.tags.get.head != "PRP" && !f.tags.get.head.contains("VB"))
+        if (inputFilter.nonEmpty) {
+          val newInputs = Map("input" -> inputFilter, "output" -> Seq())
+          val newInputMens = copyWithArgs(f, newInputs)
+          toReturn.append(newInputMens)
+        }
+      }
+      if (f.arguments.contains("output")) {
+        val outputFilter = f.arguments("output").filter(!_.label.contains("Unit") && f.tags.get.head != "PRP" && !f.tags.get.head.contains("VB"))
+        if (outputFilter.nonEmpty) {
+          val newOutputs = Map("input" -> Seq(), "output" -> outputFilter)
+          val newOutputMens = copyWithArgs(f, newOutputs)
+          toReturn.append(newOutputMens)
+        }
+      }
+    }
     toReturn ++ other
   }
 
+  def filterInputOverlaps(mentions: Seq[Mention], state: State): Seq[Mention] = {
+    val toReturn = new ArrayBuffer[Mention]()
+    for (m <- mentions) {
+      val identifierInputs = new ArrayBuffer[Mention]()
+      val phraseInputs = new ArrayBuffer[Mention]()
+      val newInputs = new ArrayBuffer[Mention]()
+      val outputs = new ArrayBuffer[Mention]()
+
+      for (arg <- m.arguments) {
+        if (arg._1 == "input") {
+          // if the argument is an input, distinguish between identifier inputs and phrase inputs
+          if (arg._2.exists(_.label == "Identifier")) {
+            identifierInputs ++= arg._2.filter(_.label.contains("Identifier"))
+            phraseInputs ++= arg._2.filterNot(_.label.contains("Identifier"))
+          } else phraseInputs ++= arg._2
+          // if there is an identifier input, check if there's an overlap between the identifier input and other phrase inputs
+          if (identifierInputs.nonEmpty) {
+            for (i <- identifierInputs) {
+              val inputNumCheck = new ArrayBuffer[Mention]
+              if (phraseInputs.nonEmpty) {
+                for (p <- phraseInputs) {
+                  newInputs.append(p)
+                  val overlappingInterval = i.tokenInterval.overlaps(p.tokenInterval)
+                  // if there's no overlap, append the identifier input to the inputNumCheck
+                  if (overlappingInterval == false) {
```

Comment:
@alicekwak we can simplify this conditional since the value you are checking is a boolean. Try replacing what you currently have with:
```scala
if (!overlappingInterval) { inputNumCheck.append(i) }
```
You can interpret the above statements as `if not overlappingInterval then inputNumCheck.append(i)`. Does this make sense?

Reply:
Yes, that is much simpler than my original code! Thank you!